src/writer.rs

```diff
@@ -190,6 +191,12 @@ where
         ..reference
     };
 
+    // The only acceptable length is 80. If different, try to set it.
+    if header.descrip.len() != 80 {
+        let descrip = header.descrip.clone();
```

Comment:
I don't think I can do that, or I don't understand how. I tried this

    let descrip = header.descrip;
    header.set_description(descrip)?;
    >>> use of partially moved value: `header`

    header.set_description(header.descrip)?;
    >>> cannot move out of `header.descrip` because it is borrowed

    header.set_description(&header.descrip)?;
    >>> type mismatch resolving `<&std::vec::Vec<u8> as std::ops::Deref>::Target == [u8]`
        expected type `std::vec::Vec<u8>`
        found type `[u8]`

The 2 first seem pretty normal to me. As for the last, I don't understand why it expects a `std::vec::Vec<u8>`.

Reply:
Oh, you are right. Moving the vector and back in does result in a partially moved value.

I suggest we add a `validate_description(&self)` method to `NiftiHeader`. It can be public or crate-public and would so expose the description validation routine:

```rust
fn validate_description(&mut self) -> Result<()> {
    let len = self.description.len();
    if len < 80 {
        self.descript.extend((len..80).map(|_| 0));
        Ok(())
    } else if len < 80 {
        Err(NiftiError::IncorrectDescriptionLength)
    }
    Ok()
}
```